x-pack/solutions/observability/plugins/slo/server/services/slo_repository.ts

```diff
@@ -178,8 +189,53 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
 
     return result.right;
   }
+
+  getDashboardsIds({
+    dashboardsRef,
+    references,
+  }: {
+    dashboardsRef?: { refId: string }[];
+    references: SavedObjectReference[];
+  }): { id: string }[] {
+    if (!dashboardsRef || dashboardsRef.length === 0) {
+      return [];
+    }
+
+    const refMap = new Map(references.map(({ name, id }) => [name, id]));
+
+    return dashboardsRef.flatMap(({ refId }) => {
+      const id = refMap.get(refId);
+      if (!id) {
+        this.logger.debug(`Invalid reference [${refId}]`);
+        return [];
+      }
+      return [{ id }];
+    });
+  }
+
+  private isSLO(slo: SLODefinition | undefined): slo is SLODefinition {
+    return slo !== undefined;
+  }
 }
 
-function toStoredSLO(slo: SLODefinition): StoredSLODefinition {
-  return sloDefinitionSchema.encode(slo);
+export function toStoredSLO(slo: SLODefinition): {
+  storedSLO: StoredSLODefinition;
+  references: SavedObjectReference[];
+} {
+  const dashboardsRef: { refId: string }[] = [];
+  const references: SavedObjectReference[] = [];
+  if (slo.artifacts?.dashboards?.length) {
+    slo.artifacts.dashboards.forEach(({ id }, index) => {
+      const refId = `dashboard-${index}`;
```

Comment:
Are we using index because we want to allow the same dashboard (id) to be referenced multiple time?

Reply:
I think it doesn't really matter what we store in the `refId` as long as it's the same as the `name` property we store in the reference.

Reply:
Exactly point is that refId should be unique and same as the name property in the reference.